todoapp/app/src/main/java/com/example/android/architecture/blueprints/todoapp/addedittask/AddEditTaskActivity.java

```diff
@@ -80,20 +82,19 @@ protected void onCreate(Bundle savedInstanceState) {
             shouldLoadDataFromRepo = savedInstanceState.getBoolean(SHOULD_LOAD_DATA_FROM_REPO_KEY);
         }
 
-        // Create the presenter
-        mAddEditTaskPresenter = new AddEditTaskPresenter(
-                taskId,
+        mViewModel = new AddEditTaskViewModel(
+                getApplicationContext(),
```

Comment:
Crazy idea; how about making the ViewModel a singleton? The dependencies it has can be injected (and cleared) from the view at an appropriate life-cycle event (onCreate/onDestroy).